noxfile.py

```diff
@@ -0,0 +1,263 @@
+}
+
+STYLE_ARGS = {
+    "pylint": [PACKAGE, "setup.py"],
+    "flake8": [PACKAGE, "setup.py", str(DOCS / "conf.py")],
+    "black": [
+        "--check",
+        PACKAGE,
+        "setup.py",
+        str(DOCS / "conf.py"),
+        "tutorials",
+        "noxfile.py",
+    ],
+}
+
+# Use absolute paths for pytest otherwise it struggles to pick up coverage info
+PYTEST_ARGS = [
+    f"--cov-config={Path('.coveragerc').resolve()}",
+    "--cov-report=term-missing",
+    f"--cov={PACKAGE}",
+    "--doctest-modules",
+    '--doctest-glob="*.rst"',
+    "-v",
+    "--pyargs",
+]
+RST_FILES = [str(p.resolve()) for p in Path("doc").glob("**/*.rst")]
+
+# Command line arguments that we're implementing. Pass them to nox after "--":
+#   nox -s test -- list-packages
+NOX_ARGS = ["skip-install", "list-packages"]
+
+
+@nox.session()
+def check(session):
+    """
+    Check code style using black, flake8, and pylint
+    """
+    install_requirements(session, ["style"])
+    list_packages(session)
+    args = list(set(session.posargs).difference(NOX_ARGS))
+    if args:
+        checks = args
+    else:
+        checks = ["black", "flake8", "pylint"]
+    for check in checks:
+        session.run(check, *STYLE_ARGS[check])
+
+
+@nox.session()
+def format(session):
+    """
+    Run black to format the code
+    """
+    install_requirements(session, ["style"])
+    session.run("black", *STYLE_ARGS["black"][1:])
+
+
+@nox.session(python=PYTHON_VERSIONS)
+def test(session):
+    """
+    Run the tests and measure coverage (using pip)
+    """
+    install_requirements(session, ["build", "run", "test"])
+    packages = build_project(session, install=True)
+    list_packages(session)
+    run_pytest(session)
+
+
+@nox.session(venv_backend="conda", python=PYTHON_VERSIONS)
+def test_conda(session):
+    """
+    Run the tests and measure coverage (using conda)
+    """
+    install_requirements(session, ["build", "run", "test"], package_manager="conda")
+    build_project(session, install=True)
+    list_packages(session, package_manager="conda")
+    run_pytest(session)
+
+
+@nox.session()
+def docs(session):
+    """
+    Build the documentation
+    """
+    install_requirements(session, ["build", "run", "docs"])
+    build_project(session, install=True)
+    list_packages(session, package_manager="pip")
+    # Generate the API reference
+    session.run(
+        "sphinx-autogen",
+        "-i",
+        "-t",
+        str(DOCS / "_templates"),
+        "-o",
+        str(DOCS / "api" / "generated"),
+        str(DOCS / "api" / "index.rst"),
+    )
+    # Build the HTML pages
+    session.run(
+        "sphinx-build",
+        "-d",
+        str(DOCS / "_build" / "doctrees"),
+        "doc",
+        str(DOCS / "_build" / "html"),
+    )
+
+
+@nox.session()
+def build(session):
+    """
+    Build source and wheel distributions for the project
+    """
+    install_requirements(session, ["build"])
+    packages = build_project(session, install=False)
```

Comment:
`packages` is assigned but never used. Don't know if you forgot to use after this line or if we can ditch the `packages` variable.